prog/weaveproxy/main.go

```diff
@@ -57,6 +59,10 @@ func main() {
 	Log.Infoln("weave proxy", version)
 	Log.Infoln("Command line arguments:", strings.Join(os.Args[1:], " "))
 
+	if withDNS {
+		Log.Warning("--with-dns option has been removed; DNS is on by default")
```

Comment:
AFAIK that's a different `--with-dns` option, on `weave run`, where the behaviour did not change.